src/hooks/tests/useNotificationModal.test.ts

```diff
@@ -0,0 +1,93 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+
+import { useNotificationModal } from "hooks";
+import { SubscriptionMethods, Trigger } from "hooks/useNotificationModal";
+import {
+  validateJira,
+  validateEmail,
+  validateSlack,
+  validateDuration,
+  validatePercentage,
+} from "utils/validators";
+
+test("Should have correctly formatted request payload after selecting options", () => {
+  const { result } = renderHook(() =>
+    useNotificationModal({
+      triggers,
+      subscriptionMethodControls,
+      resourceType: "TASK",
+      resourceId: "a task id",
+    })
+  );
+
+  act(() => {
+    result.current.setExtraFieldInputVals({ "task-duration-secs": "33" });
+  });
+  act(() => {
+    result.current.setTarget({ email: "email@email.com" });
+  });
+  act(() => {
+    result.current.setSelectedTriggerId("exceeds-duration");
+  });
+  expect(JSON.stringify(result.current.getRequestPayload())).toStrictEqual(
```

Comment:
I don't think its necessary to stringify the request payload you can probably just compare the objects.